ocaml/xapi/xapi_sr.ml

```diff
@@ -781,10 +786,23 @@ let scan ~__context ~sr =
             (* It is sufficient to just compare the refs in two db_vdis, as this
                is what update_vdis uses to determine what to delete *)
             let vdis_ref_equal db_vdi1 db_vdi2 =
-              Listext.List.set_difference (List.map fst db_vdi1)
-                (List.map fst db_vdi2)
-              = []
+              let refs1 = RefSet.of_list (List.map fst db_vdi1) in
+              let refs2 = RefSet.of_list (List.map fst db_vdi2) in
+              if RefSet.equal refs1 refs2 then
+                true
+              else
+                let log_diff str a b =
+                  RefSet.diff a b
+                  |> RefSet.elements
+                  |> List.map Ref.string_of
+                  |> String.concat " "
+                  |> debug "scan_rec: VDIs %s during scan: %s" str
```

Comment:
Would use `__FUNCTION__` to report the origin of the message but otherwise like this approach

Reply:
Yes right I think about it but function will be `log_diff` in this case. And in fact we want scan_rec. So I'm not sure about how to do it.

Reply:
or function will be the complete path maybe I'm not sure...

Reply:
it looks like it is the full path from Xapi_sr. I will fix it then :)